src/gpu/intel/gemm/jit/generator/pieces/gemm.cxx

```diff
@@ -474,6 +474,10 @@ void Generator<hw>::gemm(GEMMProblem &problem, GEMMStrategy &strategy, GEMMState
         if (checkA) emov(1 | f0[1], state.offsetA, 0, strategy, state);
         if (checkB) emov(1 | f1[1], state.offsetB, 0, strategy, state);
     }
+           cmp(1 | ge | f0[1], state.i0, state.inputs.m);
+           cmp(1 | ge | f1[1], state.j0, state.inputs.n);
+           jmpi(1 | f0[1], lKernelDone);
+           jmpi(1 | f1[1], lKernelDone);
```

Comment:
Hi @dyoussif, GEMM already skips out-of-bounds threads wherever it's safe to do so -- see [line 637](https://github.com/uxlfoundation/oneDNN/blob/11fc515f6eac5f22b0e3ae018bb96a630b83a638/src/gpu/intel/gemm/jit/generator/pieces/gemm.cxx#L637) below. 

However many situations need all the threads in a workgroup to be active:
* SLM pipeline -- hangs if not all threads enter mainloop together
* Prefetch usage -- lower performance if not all threads mainloop together